### PR TITLE
stable-v2.0: .github/zephyr: hardcode zephyr container version to v0.21.0

### PR DESCRIPTION
**Pull request by marc-hb**
The container v0.22.0 changed the toolchain locations, for instance
for APL it changed from

 `/opt/toolchains/zephyr-sdk-0.13.1/xtensa/intel_apl_adsp/...`

to

 `/opt/toolchains/zephyr-sdk-0.14.0/xtensa-intel_apl_adsp_zephyr-elf/...`

The new location is not found by the old Zephyr version fd089b361d8aebbc
that is currently hardcoded for this branch by previous SOF commit
b678a4dcaa1e. This causes the failure shown below, see real build
failure example in #5641.

The docker image v0.21.0 is the last one that has the old location.
https://hub.docker.com/r/zephyrprojectrtos/zephyr-build/tags

I suspect the name change happened in this commit:
https://github.com/zephyrproject-rtos/sdk-ng/commit/c701e23bcce86f27f3a7
```
-- Using toolchain: zephyr 0.14.0 (/opt/toolchains/zephyr-sdk-0.14.0)
-- Found dtc:
/opt/toolchains/zephyr-sdk-0.14.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc
(found suitable version "1.6.0", minimum required is "1.4.6")
  ...
  ...
CMake Error at
/workdir/zephyrproject/zephyr/cmake/compiler/gcc/target.cmake:10
-- Configuring incomplete, errors occurred!
  C compiler
/opt/toolchains/zephyr-sdk-0.14.0/xtensa/intel_apl_adsp/ \
 xtensa-intel_apl_adsp_zephyr-elf/bin/xtensa-intel_apl_adsp_zephyr-elf-gcc
      not found - Please check your toolchain installation
      Call Stack (most recent call first):
/workdir/zephyrproject/zephyr/cmake/target_toolchain.cmake:57
/workdir/zephyrproject/zephyr/cmake/app/boilerplate.cmake:599
/.../zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24
/.../zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35
```
Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by stephanosio** [comment]
@marc-hb This rather strange behaviour is observed in the Zephyr version 2.5 through 2.7 for the Xtensa targets because of the Zephyr commit https://github.com/zephyrproject-rtos/zephyr/commit/483b4ff030768af23d738c4a65076648be9e9af4.

That commit was reverted in https://github.com/zephyrproject-rtos/zephyr/commit/92a1ca61eb1ba94015a0226ac4e4e1907d480d19, so updating to the Zephyr 3.0 release should resolve this issue

**Review by greg-intel — APPROVED** [review]
Is this a hotfix?  Or long-term? 

PR is super simple, approving regardless.

**Comment by marc-hb** [comment]
Thanks @stephanosio  for the very detailed and expert info, much appreciated.

> so updating to the Zephyr 3.0 release should resolve this issue.

That's not really an option here because this is a stable branch. Instead, I cherry picked the two `zephyr/cmake/toolchain/zephyr/target.cmake` reverts e747fe73cdd67 and 92a1ca61eb1b locally and I confirm that they make this stable branch compatible with the latest SDK without breaking compatibility with 0.13.1. So cherry-picking these two reverts is a more flexible and better solution than this PR, but... this stable SOF branch can only point at upstream Zephyr for now! Bummer.

tl;dr: I know how to fix this interesting cascade of dependencies and it's not rocket-science at all (just a few more cherry-picks) but I would like the build to be fixed _first_ to stop driving blind. So let's please merge this container version hardcoding first, can be reverted later.

**Comment by marc-hb** [comment]
Un-hardcoding the Zephyr container and SDK in  this stable-v2.0 branch in new PR #5740
